packages/common/src/ProviderUtils.ts

```diff
@@ -60,6 +65,68 @@ export function createBasicProvider(nodeRetryConfig: RetryConfig[]): RetryProvid
   );
 }
 
+const KEY_TYPES = ["gckms", "mnemonic", "none"] as const;
+
+function getDefaultKeyType(): typeof KEY_TYPES[number] {
+  if (argv.network) {
+    const networkSplit = argv.network.split("_");
+    const keyType = networkSplit[networkSplit.length - 1];
+    if (KEY_TYPES.includes(keyType)) {
+      return keyType;
+    }
+  }
+  return "none";
+}
+
+function addMnemonicToProvider(
+  provider: AbstractProvider,
+  mnemonic: string = process.env.MNEMONIC ||
+    "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat",
+  numKeys = process.env.NUM_KEYS ? parseInt(process.env.NUM_KEYS) : 2,
+  keyOffset = process.env.KEY_OFFSET ? parseInt(process.env.KEY_OFFSET) : 0
+): HDWalletProvider {
+  return new HDWalletProvider(mnemonic, provider, keyOffset, numKeys);
+}
+
+function addGckmsToProvider(provider: AbstractProvider): ManagedSecretProvider {
+  const gckmsConfig = getGckmsConfig();
+  return new ManagedSecretProvider(gckmsConfig, provider, 0, gckmsConfig.length);
+}
+
+function addDefaultKeysToProvider(provider: AbstractProvider): AbstractProvider {
+  switch (getDefaultKeyType()) {
+    case "gckms":
+      return addGckmsToProvider(provider);
+    case "mnemonic":
+      return addMnemonicToProvider(provider);
+    case "none":
+    default:
+      return provider;
+  }
+}
+
+/**
+ * @notice Creates a web3 instance for a particular chain.
+ * @param chainId the chain id for the network the user wants to connect to.
+ * @returns new Web3 instance.
+ */
+export function getWeb3ByChainId(chainId: number): Web3 {
```

Comment:
nice! this is super clean. Good call on making a separate exported function. also, the notion of `nodeUrl` set to `test` is pretty good as well